In [1]:
l = 220 # px
a = 56 # deg
r = 370 # px
T = 80 # f

In [2]:
def px2m(px): # pixels to meters
    return px * .5204 / 1920

In [3]:
d = 21 # in
d = d * 2.54 / 100 # in -> cm -> m

In [5]:
import math

In [6]:
def deg(rad):
    return rad / math.pi * 180

In [7]:
ecc = deg(math.atan(px2m(r)/d))
ecc

10.647972683493427

In [8]:
rf_area = 80
rf_radius = math.sqrt(rf_area / math.pi)
rf_radius

5.046265044040321

In [9]:
inner_r = r - l/2
outer_r = r + l/2

In [10]:
inner_ecc = deg(math.atan(px2m(inner_r)/d))
outer_ecc = deg(math.atan(px2m(outer_r)/d))
print(inner_ecc, outer_ecc)

7.526119494330689 13.707208284141926


In [11]:
rf_radius2 = outer_ecc - inner_ecc
rf_radius2

6.181088789811237

In [12]:
def percent_diff(a, b):
    return 100 * abs((a - b) / ((a + b) / 2))

In [13]:
percent_diff(rf_radius, rf_radius2)

20.21533769336302

In [14]:
# period

In [24]:
w = rf_radius / (1/100)

In [25]:
dtheta = deg(math.atan(px2m(outer_r)/d) - math.atan(px2m(inner_r)/d))

In [26]:
w2 = dtheta / (T/2)
w2

0.15452721974528094

In [ ]:
percent_diff(w, w2)

199.87754910738371

possible conclusions:

* the maximal receptive field size in the MT is experimentally determined to be 6.18 deg of visual angle.
* the it is not enough for only some MT cells to solve the aperture problem; as long as there are others that aren't able to, the illusory effect exists.
* the illusory effect requires a large number of stimuli, possibly due to overcrowding and thus the aperture problem not being solved.
    * this can be observed by the fact that with only one line segment, there is no effect at all, even with longer lines.